Assets/Code/data classes/visibilty.cs

```diff
@@ -31,51 +31,51 @@ public raycastInfo(bool hit, List<planet> HitPlanets, List<satellite> HitSats, L
     }
   }
 
-  public static double distance(position p1, position p2)
+  private static double raycastMath(position position1, position position2, position obj)
   {
-     double distance = Math.Sqrt(Math.Pow((p1.x - p2.x) ,2) + Math.Pow((p1.y - p2.y) ,2) + Math.Pow((p1.z - p2.z), 2));
-     return distance;
-  }
-
-  public static double raycastMath(position position1, position position2, position obj)
-  {
-    double distance1 = distance(position1, obj);
-    double distancePercent = distance1 / distance(position1, position2);
+    double distance1 = position.distance(position1, obj);
+    double distancePercent = distance1 / position.distance(position1, position2);
 
     position pointOnLine = new position((position1.x + position2.x) * distancePercent,
     (position1.y + position2.y) * distancePercent,
     (position1.z + position2.z) * distancePercent);
 
-    double distanceFromObj = distance(pointOnLine, obj);
+    double distanceFromObj = position.distance(pointOnLine, obj);
 
     return distanceFromObj;
 
   }
-
-  //double.max
-
-  public static raycastInfo raycastHit(position p1, position p2, raycastParameters flags, int defaultRadius)
+  /// <summary> This is the main raycast function for our program and it takes in 5 arguments. The final argument if set true will return a list of all the hit objects, but if it is set to false the program will break after the first object is hit <summary> ///
+  public static raycastInfo raycast(position p1, position p2, raycastParameters flags, int defaultRadius, bool returnAllHit)
   {
-    // 100 indicates planet search
-    // 10 indicates a satellite search
-    // 1 indicates a facility search
 
     List<planet> hitPlanets = new List<planet>();
     List<facility> hitFacs = new List<facility>();
     List<satellite> hitSats = new List<satellite>();
     bool hit = false;
 
+    double p1p2Distance = position.distance(p1,p2);
+
     if ((flags & raycastParameters.planet) == raycastParameters.planet)
     {
       foreach(planet p in master.allPlanets)
       {
-        if (distance(p1, p.pos) > distance(p1,p2))
+        if (position.distance(p1, p.pos) > p1p2Distance)
         {
           double distanceFromObj = raycastMath(p1, p2, p.pos);
           if (distanceFromObj <= p.radius)
           {
-            hit = true;
-            hitPlanets.Add(p);
+            if (returnAllHit == true)
+            {
+              hit = true;
```

Comment:
the content of the if statement is repeated. take the stuff that is repeated and put it outside the loop.